discord/client.py

```diff
@@ -227,12 +248,18 @@ def __init__(
         proxy: str | None = options.pop("proxy", None)
         proxy_auth: aiohttp.BasicAuth | None = options.pop("proxy_auth", None)
         unsync_clock: bool = options.pop("assume_unsync_clock", True)
+        bucket_storage = options.pop("bucket_storage_cls", BucketStorage)(
+            options.pop("per", 1), options.pop("concurrency", 50)
+        )
+
         self.http: HTTPClient = HTTPClient(
+            bucket_storage,
             connector,
             proxy=proxy,
             proxy_auth=proxy_auth,
             unsync_clock=unsync_clock,
             loop=self.loop,
+            maximum_rate_limit_time=options.pop("maximum_rate_limit_time", -1),
```

Comment:
Putting the var definition above for code consistency
```suggestion
        rate_limit_timeout: float = options.pop("rate_limit_timeout", -1)

        self.http: HTTPClient = HTTPClient(
            bucket_storage,
            connector,
            proxy=proxy,
            proxy_auth=proxy_auth,
            unsync_clock=unsync_clock,
            loop=self.loop,
            rate_limit_timeout=rate_limit_timeout,
```